### PR TITLE
Fix issue with model layer returning connected items on private posts

### LINKED ISSUE #992: Posts set to private show full taxonomy nodes instead of `null`
When a post is set to private, the `categories` (or similar taxonomy) list will return all terms. I would expect it to return simply `null` like the rest of the post fields. I was using WP-GraphQL v0.3.8, but I am also seeing this issue on the latest version, v0.4.0.

Example query:
```
query GetPostQuery($postId: Int) {
  postBy(postId: $postId) {
    status
    title
    categories {
      nodes {
        id
        name
        slug
      }
    }
  }
}
```

Returned data of a published post with category term of "Some Category 1" selected:
```
{
  "data": {
    "postBy": {
      "status": "publish",
      "title": "Some post",
      "categories": {
        "nodes": [
          {
            "id": "Y2F0ZWdvcnk6MTQ=",
            "name": "Some Category 1",
            "slug": "some-category-1"
          }
        ]
      }
    }
  }
}
```

Returned data of same post now set to private:
```
{
  "data": {
    "postBy": {
      "status": null,
      "title": null,
      "categories": {
        "nodes": [
          {
            "id": "Y2F0ZWdvcnk6MTQ=",
            "name": "Some Category 1",
            "slug": "some-category-1"
          },
          {
            "id": "Y2F0ZWdvcnk6MQ==",
            "name": "Uncategorized",
            "slug": "uncategorized"
          }
        ]
      }
    }
  }
}
```

My categories list is currently just "Some Category 1" and "Uncategorized", so the full list is showing here. Note that this was queried from the anonymous user. This issue does not occur when querying as an admin user.

This can be replicated on a fresh install of Wordpress and WP GraphQL. Create some category terms and a post with little to no categories, then query the post for its terms.

### PR DESCRIPTION
**Pull request by CodeProKid**
### Your checklist for this pull request
Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [ ] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *our develop*.
- [ ] Make sure you are requesting to pull request from a **topic/feature/bugfix branch** (right side). Don't pull request from your master!

What does this implement/fix? Explain your changes.
---------------------------------------------------
Makes sure we don't return all terms on taxonomy connections to private posts.

Does this close any currently open issues?
------------------------------------------
closes #992 

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
It's definitely debatable that this short circuit code is in the correct place. Having the test here is useful though, and we can have a conversation around this now.

Where has this been tested?
---------------------------
**Operating System:** …

**WordPress Version:** …

### CODE REVIEW

File: src/Data/Connection/AbstractConnectionResolver.php [hunk 1]
```diff
@@ -107,6 +108,11 @@ abstract class AbstractConnectionResolver {
 	 */
 	public function __construct( $source, $args, $context, $info ) {
 
+		// Bail if the Post->ID is empty, as that indicates a private post.
+		if ( $source instanceof Post && empty( $source->ID ) ) {
+			return null;
```
Comment (jasonbahl) [inline src/Data/Connection/AbstractConnectionResolver.php]:
```suggestion
			$this->should_execute = false;
```

**Comment by CodeProKid** [comment]
@jasonbahl I don't think your change worked correctly: https://travis-ci.org/wp-graphql/wp-graphql/jobs/606779619#L2826

The test I added is failing.

File: src/Data/Connection/AbstractConnectionResolver.php [hunk 2]
```diff
@@ -107,6 +108,11 @@ abstract class AbstractConnectionResolver {
 	 */
 	public function __construct( $source, $args, $context, $info ) {
 
+		// Bail if the Post->ID is empty, as that indicates a private post.
+		if ( $source instanceof Post && empty( $source->ID ) ) {
+			$this->should_execute = false;
```
Comment (jasonbahl) [inline src/Data/Connection/AbstractConnectionResolver.php]:
```suggestion
			return null;
```

**Comment by CodeProKid** [comment]
Chatted about this during bug scrub, I think the test is just failing because it's returning `null` instead of an empty array. Just need to re-implement the change and do some testing.

There's also another ticket open that references a similar issue: https://github.com/wp-graphql/wp-graphql/issues/901 this fix _should_ apply to that ticket as well, but we should confirm with a test.

**Comment by jasonbahl** [comment]
Shoot. Didn't mean to merge this yet. 🤦‍♂

**Pull request merged**